sources/client/representation/qt/gaminglineedit.cpp

```diff
@@ -28,8 +26,32 @@ void GamingLineEdit::keyPressEvent(QKeyEvent *event)
     }
     if (event->key() == Qt::Key_F2)
     {
-        setText("OOC ");
+        ToggleOOC();
         return;
     }
     QLineEdit::keyPressEvent(event);
 }
+void GamingLineEdit::ToggleOOC(){
+    QString current_text = text();
+    if(IsOOCMessage(current_text))
+    {
+        setText(current_text.mid(current_text[3]==' '?4:3));
```

Comment:
I think, here it can be done in more safe way:
```
setText(current_text.mid(3).trim());
```
because `current_text` size can be 3 and `current_text[3]` will be an out-of-bounds access.

Reply:
`IsOOCMessage()` checks size and code `current_text[3]` never will reach if string's size will be less than 3.

So should duplicate check the size and make code more safe? Or keep as is?

Reply:
If `current_text` contains `"ooc"` or `"OOC"` string then `IsOOCMessage` will return true (because `current_text.length() >= 3`), but `current_text[3]` will try to access fourth symbol of the string (because string/array indexes start from 0). And there are no 4th symbol, because there are only 3 symbols overall in the string.